src/main/kotlin/org/elm/utils/CircularList.kt

```diff
@@ -0,0 +1,51 @@
+package org.elm.utils
+
+class CircularList<out T>(val list: List<T>) {
+
+    private val maxIndex = list.size - 1
+
+    private var index: Int = 0
+
+    fun get(): T = list[index]
+
+    fun getIndex(): Int = index
+
+    fun set(i: Int) {
+        index = i
```

Comment:
I think this needs a bounds check.

Also might want to rename the function to something like `jumpTo` as, to me, "set" implies that we are changing a value in the collection.

Reply:
Also it could make more sense, not to sift circularly but stoppen at each end... ?

Reply:
yes, then we could get rid of this entirely. And I think it would make more sense with the list view that you have.